File: cloud-manager/justice.cmp/src/main/java/com/lnjoying/justice/cmp/common/CloudStatus.java
package com.lnjoying.justice.cmp.common;

import com.lnjoying.justice.schema.entity.StatusCode;

import java.util.HashMap;
import java.util.Map;

public enum CloudStatus
{
    CREATED(0, "创建", "created"),
    IMPORTING(1, "导入中", "importing"),
    UPDATING(2, "升级中", "updating"),
    INTERNAL_TEST(3, "内测", "internal test"),

    PUBLIC_TEST(11, "公测", "public test"),
    ON_SHELF(12, "已上架", "on-shelf"),
    PRE_OFF_SHELF(13, "预下架", "pre off-shelf"),

    MAINTAINING(21, "维护中", "maintaining"),

    OFF_SHELF(31, "已下架", "off-shelf"),

    IMPORT_FILED(50, "导入失败", "import failed"),
    MAINTAIN_FILED(51, "维护失败", "maintain failed"),

    WAIT_REMOVE(81, "待删除", "wait remove"),

    REMOVING(90, "删除中", "removing"),
    REMOVE_FAILED(91, "删除失败", "remove failed"),
    REMOVED(99, "已删除", "removed");

    CloudStatus(int code, String cnDesc, String enDesc)
    {
        this.code = code;
        this.desc = new HashMap<String, String>()
        {
            {
                put("cn", cnDesc);
                put("en", enDesc);
            }
        };
    }
    
    public static CloudStatus fromCode(int code)
    {
        for (CloudStatus current : CloudStatus.values())
        {
            if (current.getCode() == code)
            {
                return current;
            }
        }
        throw new RuntimeException("Invalid error code: " + code);
    }

    private int code;
    
    private Map<String, String> desc;

    public int getCode() {
        return code;
    }

    public Map<String, String> getDesc()
    {
        return desc;
    }

    public StatusCode toStatusCode()
    {
        return new StatusCode(this.code, this.desc);
    }
}

File: web/src/utils/status/cloudStatus.js
import i18n from "@/utils/i18n/index.js";

const codeMessageMap = {
  0: i18n.t("statusAndType.create"), // created
  1: i18n.t("statusAndType.importing"), // importing
  2: i18n.t("statusAndType.upgrading"), // updating
  3: i18n.t("statusAndType.internalTest"), // internal test
  11: i18n.t("statusAndType.publicTest"), // public test
  12: i18n.t("statusAndType.onShelf"), // on-shelf
  13: i18n.t("statusAndType.preOffShelf"), // pre off-shelf
  21: i18n.t("statusAndType.maintaining"), // maintaining
  31: i18n.t("statusAndType.offShelf"), // off-shelf
  50: i18n.t("statusAndType.importFailed"), // import failed
  51: i18n.t("statusAndType.maintainFailed"), // maintain failed
  90: i18n.t("statusAndType.deleting"), // removing
  91: i18n.t("statusAndType.deleteFailed"), // remove failed
  99: i18n.t("statusAndType.deleted"), // removed
};
const codeMessageTagMap = {
  0: "", // created
  1: "", // importing
  2: "", // updating
  3: "success", // internal test
  11: "success", // public test
  12: "success", // on-shelf
  13: "warning", // pre off-shelf
  21: "warning", // maintaining
  31: "warning", // off-shelf
  50: "danger", // import failed
  51: "danger", // maintain failed
  90: "warning", // removing
  91: "danger", // remove failed
  99: "danger", // removed
};
const showCloudStatusMessage = (code, type) => {
  if (type == "tag") {
    return codeMessageTagMap[JSON.stringify(code)] != undefined
      ? codeMessageTagMap[JSON.stringify(code)]
      : "info";
  } else {
    return (
      codeMessageMap[JSON.stringify(code)] ||
      i18n.t("statusAndType.unknownStatus")
    );
  }
};

export default showCloudStatusMessage;
